frontend/components/generic/CustomStepper.tsx

```diff
@@ -0,0 +1,66 @@
+import { Step, StepConnector, stepConnectorClasses, StepIconProps, StepLabel, Stepper } from '@mui/material';
+import { styled } from '@mui/system';
+import CheckIcon from '@mui/icons-material/Check';
+import CircleIcon from '@mui/icons-material/Circle';
+
+interface Props {
+  activeStep?: number;
+  stepLabels: string[];
+}
+
+const CustomStepIcon = (props: StepIconProps) => {
+  const { active, completed } = props;
+
+  const StepIconRoot = styled('div')<{
+    ownerState: { completed?: boolean; active?: boolean };
+  }>(({ theme, ownerState }) => ({
+    backgroundColor: theme.palette.neutral.light,
+    zIndex: 1,
+    color: '#FFFFFF',
+    display: 'flex',
+    borderRadius: '50%',
+    justifyContent: 'center',
+    alignItems: 'center',
+    padding: '4px',
+    ...(ownerState.active && {
+      backgroundImage: `linear-gradient(45deg, ${theme.palette.primary.main} 30%, ${theme.palette.secondary.main} 90%)`,
+    }),
+    ...(ownerState.completed && {
+      backgroundImage: `linear-gradient(45deg, ${theme.palette.primary.main} 30%, ${theme.palette.secondary.main} 90%)`,
+    }),
```

Comment:
Yea because each MUI stepper state has a different style so if you want to change the style for that state, you need to specify it, it doesn't get carried over